apis/nucleus/src/components/listbox/assets/get-list-sizes.js

```diff
@@ -41,6 +41,16 @@ export default function getListSizes({ layout, width, height, listCount, count,
     }
   }
 
+  columnCount = dataLayout === 'singleColumn' ? 1 : columnCount;
+  rowCount = dataLayout === 'singleColumn' ? count : rowCount;
+  const maxRowCount = layoutOptions.dense ? 838000 : 577000; // Styling breaks on items above this number: https://github.com/bvaughn/react-window/issues/659
+  rowCount = Math.min(rowCount, maxRowCount);
+  const maxColumnCount = Math.floor(33550000 / columnWidth);
+  columnCount = Math.min(columnCount, maxColumnCount);
```

Comment:
Should we add a comment about the max columns as well? Or where does that number come from? (A lot of columns though 😮)

Reply:
I named that number in a const, to clarify what is is. Also added a comment to the limitation bug.